gnucash/gnome/assistant-stock-transaction.cpp

```diff
@@ -207,10 +207,10 @@ static const TxnTypeVec long_types
         FieldMask::ENABLED_CREDIT,         // dividend_amt
         FieldMask::DISABLED,               // capg_amt
         // Translators: this is a stock transaction describing
-        // dividend issued to holder, and is reinvested. Some
+        // dividend issued to holder, and it may be reinvested. Some
         // dividends are distributed as cash.
-        N_("Dividend reinvestment (with remainder)"),
-        N_("Company issues dividend which is reinvested. Some dividends are paid to holder")
+        N_("Dividend reinvestment"),
+        N_("Company issues dividend, which may be reinvested. Non-reinvested dividends are paid out as cash.")
```

Comment:
@CDB-Man If this option will take care of both "dividend reinvestment" and "dividend reinvestment with remainer", then the `cash_amount` in the truth table will then need to allow ENABLE_DEBIT and ZERO. So, a few lines above the strings will need to be changed as follows:

```patch
@@ -201,7 +201,7 @@ static const TxnTypeVec long_types
     {
         FieldMask::ENABLED_DEBIT,          // stock_amt
         FieldMask::ENABLED_DEBIT,          // stock_val
-        FieldMask::ENABLED_DEBIT,          // cash_amt
+        FieldMask::ENABLED_DEBIT | FieldMask::ALLOW_ZERO,          // cash_amt
         FieldMask::ENABLED_DEBIT | FieldMask::ALLOW_ZERO,          // fees_amt
         true,                   // fees_capitalize
         FieldMask::ENABLED_CREDIT,         // dividend_amt
```

The other UX issue is that the updated transaction description `"Company issues dividend, which may be reinvested. Non-reinvested dividends are paid out as cash."` is very much clearer, however either

1. the "accounting"-type definition could/should either be extended to include the stock-assistant UI hints, e.g. `"Company issues dividend, which may be reinvested to purchase more units. Non-reinvested dividends are paid out as cash. The dividend amount is input in the Dividend tab. If there are no non-reinvested dividends, the cash amount in the assistant will be zero."`. As you can see it gets unwieldly very quickly. The advantage is there's only 1 long string per transaction type.

2. the "accounting"-type definition stays as is, and the assistant pages' descriptions may by individuali[s|z]ed with custom strings. In this dividend reinvestment, the cash description could be "Please input the non-reinvested cash remainder. If there are no non-reinvested dividends, please leave blank or input zero". This is cleaner, but generate a NxM number of strings which can annoy translators.

3. We can ignore these UX concerns altogether (for now) :)

Reply:
1. The cash flag is changed as suggested.
2. See 6ceee1b for a compromise on the definition text.  I really don't want to create for ourselves a situation with NxM future strings to maintain.  Each of the individual steps are basically identical, so any variation should be front loaded.